app_server/app/policies/Authenticator.js

```diff
@@ -190,8 +190,15 @@ Class.validateAccount = function (server, session) {
       return null;
     }
 
-    return session.username === cached.username &&
-           session.password === cached.password;
+    var isAccountValidated = session.username === cached.username &&
+                             session.password === cached.password;
+
+    if (!isAdminScope(session.scope)) {
+      return isAccountValidated;
+    } else {
+      return isAccountValidated && session.csrfToken &&
+             session.csrfToken === cached.encryptedPassword;
```

Comment:
It seems the presence of a custom header itself is enough to prevent usual CSRF attacks regardless of its content, actually.

I was thinking of what could I put as its content without the need to store more data in the database and figured the encrypted password would be secret enough.
That said, it contains nothing verifiable linked to the session, so I'm switching to the cookie instead. This means losing the HttpOnly protection, but that's not really anywhere near foolproof either.